File: skyspacefivefiftyfive_ui/src/services/ItineraryService.js
import http from "../http-common"; 

class ItineraryService {
  getAllItinerarys(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/itinerary/itinerarys`, searchDTO);
  }

  get(itineraryId) {
    return this.getRequest(`/itinerary/${itineraryId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/itinerary?field=${matchData}`, null);
  }

  addItinerary(data) {
    return http.post("/itinerary/addItinerary", data);
  }

  update(data) {
  	return http.post("/itinerary/updateItinerary", data);
  }
  
  uploadImage(data,itineraryId) {
  	return http.postForm("/itinerary/uploadImage/"+itineraryId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ItineraryService();

File: src/main/java/com/skyspacefivefiftyfive/service/ItineraryService.java
package com.skyspacefivefiftyfive.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspacefivefiftyfive.domain.Itinerary;
import com.skyspacefivefiftyfive.dto.ItineraryDTO;
import com.skyspacefivefiftyfive.dto.ItinerarySearchDTO;
import com.skyspacefivefiftyfive.dto.ItineraryPageDTO;
import com.skyspacefivefiftyfive.dto.ItineraryConvertCriteriaDTO;
import com.skyspacefivefiftyfive.service.GenericService;
import com.skyspacefivefiftyfive.dto.common.RequestDTO;
import com.skyspacefivefiftyfive.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ItineraryService extends GenericService<Itinerary, Integer> {

	List<Itinerary> findAll();

	ResultDTO addItinerary(ItineraryDTO itineraryDTO, RequestDTO requestDTO);

	ResultDTO updateItinerary(ItineraryDTO itineraryDTO, RequestDTO requestDTO);

    Page<Itinerary> getAllItinerarys(Pageable pageable);

    Page<Itinerary> getAllItinerarys(Specification<Itinerary> spec, Pageable pageable);

	ResponseEntity<ItineraryPageDTO> getItinerarys(ItinerarySearchDTO itinerarySearchDTO);
	
	List<ItineraryDTO> convertItinerarysToItineraryDTOs(List<Itinerary> itinerarys, ItineraryConvertCriteriaDTO convertCriteria);

	ItineraryDTO getItineraryDTOById(Integer itineraryId);







}





